src/platform/mod.rs

```diff
@@ -8,8 +8,9 @@
 // except according to those terms.
 
 mod os {
-    #[cfg(all(not(feature = "force-inprocess"), target_os = "linux"))]
-    include!("linux/mod.rs");
+    #[cfg(all(not(feature = "force-inprocess"), any(target_os = "linux",
+                                                    target_os = "freebsd")))]
```

Comment:
Personally, I'd add the `freebsd` case only right before or along with actually adding FreeBSD-specific code, after the generic portability improvements... But this way is fine too I guess :-)